### PR TITLE
Fix kubelet to not accept negative eviction (hard, soft) thresholds

### PR DESCRIPTION
**Pull request by aveshagarwal**
@kubernetes/rh-cluster-infra @derekwaynecarr

### CODE REVIEW

**Comment by aveshagarwal** [comment]
Updated with unit tests.

File: pkg/kubelet/eviction/helpers.go [hunk 1]
```diff
@@ -157,6 +157,10 @@ func parseThresholdStatement(statement string) (Threshold, error) {
 	if err != nil {
 		return Threshold{}, err
 	}
+	if quantity.Sign() < 0 {
+		return Threshold{}, fmt.Errorf("eviction threshold %v cannot be negative: %s", signal, quantity.String())
```
Comment (sttts) [inline pkg/kubelet/eviction/helpers.go]: nit: `%s` leads to a call to `String()` anyway.

**Comment by sttts** [comment]
One nit, otherwise lgtm.

File: pkg/kubelet/eviction/helpers.go (hunk 1 above)
Comment (aveshagarwal) [inline pkg/kubelet/eviction/helpers.go]: That is my understanding too. Somehow when I was testing it yesterday, it did not work this way, I am not sure why, so I called it explicitly. I think I will test it again today. Thanks anyway.

**Comment by sttts** [comment]
lgtm. Why does the bot apply do-not-merge?

**Comment by ncdc** [comment]
@apelisse is there a bug in the bot re do-not-merge and retest-not-required-docs-only?

**Pull request merged**